acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -301,6 +295,39 @@ impl<'backend, B: BlackBoxFunctionSolver> ACVM<'backend, B> {
             }
         }
     }
+
+    fn solve_brillig_opcode(
+        &mut self,
+    ) -> Result<Option<ForeignCallWaitInfo>, OpcodeResolutionError> {
+        let Opcode::Brillig(brillig) = &self.opcodes[self.instruction_pointer] else {
+            unreachable!("Not executing a Brillig opcode");
+        };
+        let witness = &mut self.witness_map;
+        if BrilligSolver::<B>::should_skip(witness, brillig)? {
+            BrilligSolver::<B>::zero_out_brillig_outputs(witness, brillig).map(|_| None)
+        } else {
+            let mut solver = match self.brillig_solver.take() {
+                None => {
+                    BrilligSolver::new(witness, brillig, self.backend, self.instruction_pointer)?
+                }
+                Some(solver) => solver,
+            };
+            match solver.solve()? {
+                BrilligSolverStatus::ForeignCallWait(foreign_call) => {
+                    _ = self.brillig_solver.insert(solver);
+                    Ok(Some(foreign_call))
```

Comment:
```suggestion
                    self.brillig_solver = Some(solver);
```

This would be a more idiomatic way to do this.